detectable/src/main/java/com/blackduck/integration/detectable/detectables/yarn/YarnTransformer.java

```diff
@@ -210,42 +212,66 @@ private LazyBuilderMissingExternalIdHandler getLazyBuilderHandler(List<NameVersi
     }
-    
-    private void addRootDependenciesForProjectOrWorkspace(YarnLockResult yarnLockResult, NullSafePackageJson projectOrWorkspacePackageJson, ExternalIdDependencyGraphBuilder graphBuilder) throws MissingExternalIdException {
-        addRootDependenciesToGraph(graphBuilder, projectOrWorkspacePackageJson.getDependencies(), yarnLockResult.getWorkspaceData());
-        if (yarnDependencyTypeFilter.shouldInclude(YarnDependencyType.NON_PRODUCTION)) {
-            addRootDependenciesToGraph(graphBuilder, projectOrWorkspacePackageJson.getDevDependencies(), yarnLockResult.getWorkspaceData());
-        }
-    }
 
-    private void addRootDependenciesToGraph(ExternalIdDependencyGraphBuilder graphBuilder, Map<String, String> rootDependenciesToAdd, YarnWorkspaces workspaceData) {
-        for (Map.Entry<String, String> rootDependency : rootDependenciesToAdd.entrySet()) {
-            Optional<YarnWorkspace> dependencyWorkspace = workspaceData.lookup(rootDependency.getKey(), rootDependency.getValue());
-            if (dependencyWorkspace.isPresent()) {
-                logger.trace("Omitting dependency {}/{} because it's a workspace", rootDependency.getKey(), rootDependency.getValue());
+    private LazyBuilderMissingExternalIdHandler getLazyBuilderHandler(List<NameVersion> externalDependencies, Map<LazyId, NameVersion> yarnLockDependencies) {
+        return (dependencyId, lazyDependencyInfo) -> {
+            if (yarnLockDependencies.containsKey(dependencyId)) {
+                NameVersion matchedDependency = yarnLockDependencies.get(dependencyId);
+                return generateComponentExternalId(matchedDependency.getName(), matchedDependency.getVersion());
+            }
+            Optional<NameVersion> nameVersionOptional;
+            if (!unMatchedDependencies.containsKey(dependencyId)) {
+                String dependencyIdClean = dependencyId.toString().replace("\\/","/");
+                nameVersionOptional = getNameVersion(dependencyIdClean);
+                logger.warn("No standard NPM package identification details are available for '{}' from the yarn.lock file", dependencyIdClean);
+                unMatchedDependencies.put(dependencyId, nameVersionOptional);
             } else {
-                LazyId stringDependencyId = generateComponentDependencyId(rootDependency.getKey(), rootDependency.getValue());
-                logger.debug("Adding root dependency to graph: stringDependencyId: {}", stringDependencyId);
-                graphBuilder.addChildToRoot(stringDependencyId);
+                nameVersionOptional = unMatchedDependencies.get(dependencyId);
             }
+            return generateComponentExternalId(nameVersionOptional.get().getName(), nameVersionOptional.get().getVersion());
+        };
+    }
+    
+    private Optional<NameVersion> getNameVersion(String dependencyIdString) {
+        int start, mid, end;
+        String name;
+        if ((start = dependencyIdString.indexOf("STRING\",\"")) > -1
```

Comment:
I don't recognize these `STRING` and `NAME_VERSION` tokens.

Could you shed some light on where those are coming from and what possible values are?